package/MDAnalysis/auxiliary/EDR.py

```diff
@@ -0,0 +1,435 @@
+system, such as energies, temperature, or pressure.
+
+pyedr_ is a Python package that reads EDR binary files and returns them
+human-readable form as a dictionary of NumPy arrays. It is used by the EDR
+auxiliary reader to parse EDR files. As such, a dictionary with string keys and
+numpy array values is loaded into the :class:`EDRReader`. It is basically a
+Python-based version of the C++ code in GROMACS_.
+
+The EDR auxiliary reader takes the output from pyedr and makes this data
+available within MDAnalysis. The usual workflow starts with creating an
+EDRReader and passing it the file to read as such::
+
+    import MDAnalysis as mda
+    aux = mda.auxiliary.EDR.EDRReader(some_edr_file)
+
+The newly created `aux` object contains all the data found in the EDR file. It
+is stored in the :attr:`.data_dict` dictionary, which maps the names GROMACS
+gave each data entry to a NumPy array that holds the relevant data. These
+GROMACS-given names are stored in and available through the :attr:`.terms`
+attribute. In addition to the numeric data, the new `EDRReader` also stores the
+units of each entry in the :attr:`.data_dict` dictionary in its
+:attr:`.unit_dict` dictionary.
+
+.. warning::
+    Units are converted to `MDAnalysis base units`_ automatically unless
+    otherwise specified. However, not all unit types have a defined base unit
+    in MDAnalysis. (cf. :data:`MDAnalysis.units.MDANALYSIS_BASE_UNITS`).
+    Pressure units, for example, are not currently defined, and
+    will not be converted. This might cause inconsistencies between units!
+    Conversion can be switched off by passing `convert_units=False` when the
+    EDRReader is created::
+
+        aux = mda.auxiliary.EDR.EDRReader(some_edr_file, convert_units=False)
+
+
+Standalone Usage of the EDRReader
+---------------------------------
+
+The :class:`.EDRReader` can be used to access the data stored in EDR files on
+its own, without association of data to trajectory frames.
+This is useful, for example, for plotting. The data for a single term, a list
+of terms, or for all terms can be returned in dictionary form. "Time" is always
+returned in this dictionary to make plotting easier::
+
+
+    temp = aux.get_data("Temperature")
+    plt.plot(temp["Time"], temp["Temperature"])
+
+    some_terms = aux.get_data(["Potential", "Kinetic En.", "Box-X"])
+    plt.plot(some_terms["Time"], some_terms["Potential"])
+
+    all_terms = aux.get_data()
+    plt.plot(all_terms["Time"], all_terms["Pressure"])
+
+Adding EDR data to trajectories
+-------------------------------
+
+Like other AuxReaders, the :class:`.EDRReader` can attach its data to a
+trajectory by associating it to the appropriate time steps.
+In general, to add EDR data to a trajectory, one needs to provide two
+arguments.
+
+.. note::
+    The following will change with the release of MDAnalysis 3.0, which is
+    scheduled for September 2023. From then on, the order of these two
```

Comment:
Please don't add specific dates for releases in the docs. We never know what might happen and I prefer knowing exactly where our public facing schedule is (which right now is a work-in progress blog post).